### DISCUSSION TITLE
Maintainership status for Workspacer up ahead

### DISCUSSION THREAD

**Discussion by josteink** [rickbutton/workspacer#485] [Announcements]
Hey everyone.

**A quick, dramatic 3-point executive summary:**

* I'm stepping down as a maintainer.
* If there are anyone around willing to pick up the project, that's great! Let us know!
* If *not* we should probably be honest about it, and add a deprecation/obsoletion warning in the readme and on the website.

**More in depth...**

I haven't really used Windows as my "main" OS in a few years now, and it's not about to change. That means I don't currently need (nor have needed) Workspacer for anything for quite a while.

That makes me a pretty bad candidate for being "the" maintainer.

I've stuck around as a maintainer none the less, because approving easy patches is not that much work... But I think the time has come for me to step down, as I really don't have any investment in this project anymore. 

Right now, I don't see much activity from anyone at all. I don't know if that's because you've left for different projects, or because Workspacer currently covers all your needs well enough.

Do you think you can take a more active role? Do you depend on this software and would be sad to see it left to (bit)rot? Or have you left entirely?

What's the status of you guys out there? And what should be the status of the workspacer project from now on?

Any feedback appreciated.

CC: @rickbutton @Aareon @dalyIsaac @wiredmatt @N1x0 @sitiom @tlsomers

**Comment by sitiom** [comment]
> I haven't really used Windows as my "main" OS in a few years now, and it's not about to change. That means I don't currently need (nor have needed) Workspacer for anything for quite a while.

Same here.

> Right now, I don't see much activity from anyone at all. I don't know if that's because you've left for different projects, or because Workspacer currently covers all your needs well enough.

And if I ever use Windows, I've switched to using [komorebi](https://github.com/LGUG2Z/komorebi) as my tiling wm of choice. I've been maintaining its [application rules](https://github.com/LGUG2Z/komorebi-application-specific-configuration) for the past year, so it should be quite apparent that I've moved away from using workspacer 😅

**Comment by wiredmatt** [comment]
IMO Workspacer has been good enough for a while, in fact, I haven't updated it since last year, which means this is already mature, usable, and feature complete. Great work everyone!

**Comment by Aareon** [comment]
I'm unfortunately not versed enough with the language and libraries to be
able to take over such a project. Thanks for dedicating so much of your
time and expertise Jostein. I remain nearby if ever there is anything I can
do. In the meantime, I wish you the best of luck with your future projects.

**Comment by rickbutton** [comment]
@josteink first, a massive thank you for managing this project. Workspacer would probably have died an early death had you not reached out and offered to help maintain it. I can't thank you enough; truly a great example of open source magic.

I'm in a similar boat with workspacer, in that I don't use it day-to-day any more and as such find it hard to dedicate time to maintaining this, and am also probably not the best steward of the project as a non-user. 

If someone who is interested in helping review PRs/issues wants to reach out, please feel free, especially if you are already/have contributed to the project or participated in discussions. In the mean time, @josteink don't feel obligated to do anything more than you wish to, Workspacer is in a good state as far as I can tell, and short of Windows breaking changes to explorer/dwm in the future, I imagine it will behave in the medium/long term.

I think that it is probably a good idea to put some kind of notice/warning on the README that indicates that workspacer is 1. not actively being maintained and 2. looking for a maintainer (although probably with a bit of a disclaimer, `workspacer` is a native executable that I wouldn't feel comfortable putting in just -anyone's- hands to possibly do bad things with).

In the mean time, I'll pin this issue to give it some more visibility. 

Thanks again to everyone who has contributed to workspacer, I want to give a shout out to the many people who've contributed code, experience, or feedback:
- @josteink 
- @sitiom 
- @dalyIsaac 
- @N1x0 
- @tlsomers 
- @BeeryShklar 
- @haoxiangliew 
- @denjukebox 
- @Janthor 
- @Aareon 
- @wiredmatt
- And many others: https://github.com/workspacer/workspacer/graphs/contributors
  - I couldn't possibly include everyone here, so apologies if you were missed, I appreciate your contribution to this tiny corner of the internet!!!

**Comment by Aareon** [comment]
> workspacer is a native executable that I wouldn't feel comfortable putting in just -anyone's- hands to possibly do bad things with).

I would like to offer my services in vetting new PRs from any new maintainers as it would be an absolute shame if anything malicious made it into the codebase.

**Comment by dalyIsaac** [comment]
A massive thank you to @rickbutton for creating Workspacer and to @josteink for shepherding it for the past few years. I haven't been using Workspacer for a while, as I've been working on a [new window manager (Whim)](https://github.com/dalyIsaac/Whim) which is very much inspired by Workspacer, but tries to extend it past some of the limitations I've encountered with it. Consequently, I felt it may not be entirely appropriate for me to actively maintain Workspacer.

In terms of ongoing support, I agree with @rickbutton that putting a warning on the README is probably best. If the project isn't archived, I'd be open to keeping an eye on any severe issues that come up, but I wouldn't actively work on it.

A big thanks again to everyone who has contributed to Workspacer over the years. It's been a great project to work on and I've learned a lot from it.

**Comment by tlsomers** [comment]
Thanks a lot for @rickbutton and @josteink for maintaining the project over the past years. Sadly I'm in the same boat as most of the other people, as I've not been using Windows much.

Thanks a lot to everyone else who has been contributing to workspacer as well, and good luck with whatever projects you decide to pick up in the future.

**Comment by shlemiel** [comment]
I think the project is already mature enough.

I previously tried almost all other TWMs made for windows (komorebi, bug.n, etc.) but none of them was solid enough. Note that most of them were ahk based, which is not good enough base for tiling window manager IMO.

Anyway, I personally wrote small code to put cursor in the center of the newly focused window.

```
public class CenterizerPlugin : IPlugin
{
    private static Logger Logger = Logger.Create();
    private IConfigContext _context;

    public CenterizerPlugin() { }

    public void AfterConfig(IConfigContext context)
    {
        _context = context;

        _context.Windows.WindowFocused += WindowFocused;
    }

    private void WindowFocused(IWindow window)
    {
        var location = window.Location;
        var centerX = location.X + location.Width / 2;
        var centerY = location.Y + location.Height / 2;
        System.Windows.Forms.Cursor.Position = new System.Drawing.Point(centerX, centerY);
    }
}
```
